packages/common/gotrue.ts

```diff
@@ -1,42 +1,30 @@
-import { GoTrueClient } from '@supabase/gotrue-js'
+import { GoTrueClient, navigatorLock } from '@supabase/gotrue-js'
 
 export const STORAGE_KEY = process.env.NEXT_PUBLIC_STORAGE_KEY || 'supabase.dashboard.auth.token'
 export const AUTH_DEBUG_KEY =
   process.env.NEXT_PUBLIC_AUTH_DEBUG_KEY || 'supabase.dashboard.auth.debug'
+export const AUTH_NAVIGATOR_LOCK_KEY =
+  process.env.NEXT_PUBLIC_AUTH_NAVIGATOR_LOCK_KEY || 'supabase.dashboard.auth.navigatorLock.enabled'
 
 const debug =
   process.env.NEXT_PUBLIC_IS_PLATFORM === 'true' &&
   globalThis &&
   globalThis.localStorage &&
   globalThis.localStorage.getItem(AUTH_DEBUG_KEY) === 'true'
 
-let customFetch = globalThis.fetch
-
-if (debug) {
-  customFetch = async (a, b) => {
-    const el = (event: any) => {
-      event.preventDefault()
-      console.log('GoTrue fetch in progress detected in beforeunload!', a, b)
-    }
-
-    try {
-      if (document) {
-        document.addEventListener('beforeunload', el, { capture: true })
-      }
-
-      return await globalThis.fetch(a, b)
-    } finally {
-      if (document) {
-        document.removeEventListener('beforeunload', el, { capture: true })
-      }
-    }
-  }
-}
+const navigatorLockEnabled =
+  process.env.NEXT_PUBLIC_IS_PLATFORM === 'true' &&
+  globalThis &&
+  globalThis.localStorage &&
+  globalThis.localStorage.getItem(AUTH_NAVIGATOR_LOCK_KEY) === 'true'
 
 export const gotrueClient = new GoTrueClient({
   url: process.env.NEXT_PUBLIC_GOTRUE_URL,
   storageKey: STORAGE_KEY,
   detectSessionInUrl: true,
   debug,
-  fetch: customFetch,
+  lock:
+    (debug || navigatorLockEnabled) && globalThis.navigator && globalThis.navigator.locks
+      ? navigatorLock
+      : undefined,
```

Comment:
Should we still pass `fetch: globalThis.fetch`? At least before this change we passed in globalThis.fetch, even with debug mode disabled

Reply:
gotrue-js uses the window fetch by default. I added this to try something but turns out it's not that useful so I'm removing the code. This is the change.

https://github.com/supabase/supabase/commit/5a98d79715ff0f2e6df7eede6e926a767f4a3a07